rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -3131,21 +3131,14 @@ namespace rsx
 				return;
 
 			const auto memory_end = memory_address + memory_range;
-			u32 sync_address = 0;
 
 			for (auto It = m_pending_writes.crbegin(); It != m_pending_writes.crend(); ++It)
 			{
 				if (It->sink >= memory_address && It->sink < memory_end)
 				{
-					sync_address = It->sink;
-					break;
+					update(ptimer, It->sink);
```

Comment:
Notice how the list is being traversed in reverse? update will update all of them upto this one in series. If you get the last one, all the previous ones will have been implicitly written as well.

Reply:
right.